Service/ImageService.cs

```diff
@@ -18,5 +22,31 @@ public string ImageBase64(byte[] imageData)
 
             return $"data:image/png;base64,{Convert.ToBase64String(imageData)}";
         }
+
+        public async Task<List<Image>> UploadImagesAsync(List<IBrowserFile> files, string moveName)
```

Comment:
Patrz jak ładniej można napisać z LINQ - using Microsoft.AspNetCore.Components.Forms;
using mPole.Data.Models;
using mPole.Interface;

namespace mPole.Services
{
    public class ImageService : IImageService
    {
        private readonly IImageRepository _imageRepository;
        private readonly List<string> allowedFormats = new List<string> { "image/jpeg", "image/png", "image/gif" };

        public ImageService(IImageRepository imageRepository, IWebHostEnvironment environment)
        {
            _imageRepository = imageRepository;
        }

        public string ImageBase64(byte[] imageData)
        {
            return $"data:image/png;base64,{Convert.ToBase64String(imageData)}";
        }

        public async Task<List<Image>> UploadImagesAsync(List<IBrowserFile> files, string moveName)
        {
            var uploadedImages = await Task.WhenAll(
                files
                    .Where(file => allowedFormats.Contains(file.ContentType)) // Filter only allowed formats
                    .Select(async file =>
                    {
                        using var memoryStream = new MemoryStream();
                        await file.OpenReadStream().CopyToAsync(memoryStream);

                        return new Image
                        {
                            Name = moveName,
                            ImageData = memoryStream.ToArray()
                        };
                    })
            );

            return uploadedImages.ToList(); // Convert the result to a List<Image>
        }
    }
}
 - zweryfikuj bo to z ChatBota - ale nie zawsze foreach jest git, czasami jest Git bo LINQ sobie nie poradzi - ale w wielu sytuacjach LINQ jest szybsze